pkg/registry/generic/registry/store.go

```diff
@@ -40,6 +41,10 @@ import (
 	"github.com/golang/glog"
 )
 
+// EnableGarbageCollector affects the handling of Update and Delete requests. It
+// must be synced with the corresponding flag in kube-controller-manager.
+var EnableGarbageCollector bool
```

Comment:
At the very least, though, make a copy of this variable in the Store object, default it to this value at construction time. That way this will at least be testable.